From d57b143085bfa66319a4014f66882ecfbf12b862 Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Tue, 29 Jun 2021 19:38:56 +0300
Subject: [PATCH] [SW-43475] greco: support lzo decompression

Application like DSV is big in size (~3.7 MB).
This is bigger than the size of image supported to
load from host(via PCIe). So DSV image needs to be compressed
to fit into the size requirement set in preboot.
So removed the stub function such that common weak function
implementation gets effective which does the decompression
lzo compressed images.

Change-Id: I457ab7fbde3052c56dc0e797c8376297c1780e68
Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
---

diff --git a/arch/greco/common.c b/arch/greco/common.c
index dc436f2..9310b79 100644
--- a/arch/greco/common.c
+++ b/arch/greco/common.c
@@ -703,13 +703,6 @@
 		memcpy((void *)dst, buf, len);
 }
 
-int preboot_decompress_image(u64 dst, const void *buf, int len,
-			     const char *arch, const char *compression)
-{
-	preboot_puts("Compression is not supported\n");
-	return 1;
-}
-
 void ppboot_cp_preboot(u64 loadaddr, u32 offset, u32 size, char *name)
 {
 	preboot_puts_hex_raw("Preboot loadaddr: ",
diff --git a/arch/greco_arc_gnu/common.c b/arch/greco_arc_gnu/common.c
index ea31617..602605e 100644
--- a/arch/greco_arc_gnu/common.c
+++ b/arch/greco_arc_gnu/common.c
@@ -182,13 +182,6 @@
 	memcpy((void *)arc_dst, buf, len);
 }
 
-int preboot_decompress_image(u64 dst, const void *buf, int len,
-			     const char *arch, const char *compression)
-{
-	preboot_puts("Compression is not supported\n");
-	return 1;
-}
-
 void preboot_start_image(u64 startaddr)
 {
 	void (*image_entry)(void);
